### PR TITLE
Qol reorganize and update all the squads vendors.

### PR DESCRIPTION
**Pull request by Huffie56**
# About the pull request
1-more categories
2-making the same organization category and order in all the squad marines vendors.(easier to navigate when changing roles.)
3-remove a bunch of holster being sell when you can choose them for free...

# Explain why it's good for the game
1-making it easier for the players to navigate in squads vendors.
2-remove trap item that are being sold.
3-adding items that could be sold...

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
qol: reorganize and update all the squads vendors to add new categories(helmet optics, pamphlets, clothing items).
balance: remove hostler because it was available for free(for medic, squad-tl,.
balance: added possibility to buy M4 Pattern Armor for rifleman.
balance: added possibility to buy USCM Radio Telephone Pack for squad medic, engi and specialist.
balance: added possibility to buy Welding Visor for medic, SL, rifleman, smartgunner and specialist.
/:cl:

### CODE REVIEW

**Comment by Huffie56** [comment]
VERY WIP.

**Comment by BeagleGaming1** [comment]
Could you list what items are being added and removed instead of it just being "for consistency"?

**Comment by Huffie56** [comment]
> 

yes

**Comment by Huffie56** [comment]
i am in need of head dev directions concerning some items that are available freely on the squad prep or in medbay but are costing points in vendor.
wish confuse me greatly 

> Could you list what items are being added and removed instead of it just being "for consistency"?

done

**Comment by Huffie56** [comment]
i am in need of head dev directions concerning some items that are available freely on the squad prep or in medbay but are costing points in vendor.
wish confuse me greatly

**Comment by Huffie56** [comment]
https://forum.cm-ss13.com/t/trying-to-make-sense-and-add-consistency-to-item-price-and-availability/4026

File: code/game/machinery/vending/vendor_types/squad_prep/squad_leader.dm [hunk 1]
```diff
@@ -85,6 +92,9 @@ GLOBAL_LIST_INIT(cm_vending_gear_leader, list(
 		list("SU-6 Smart Pistol", 12, /obj/item/storage/box/guncase/smartpistol, null, VENDOR_ITEM_REGULAR),
 		list("M79 Grenade Launcher", 18, /obj/item/storage/box/guncase/m79, null, VENDOR_ITEM_REGULAR),
 
+		list("PAMPHLETS", 0, null, null, null),
+		list("Powerloader Certification", 45, /obj/item/pamphlet/skill/powerloader, null, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_leader.dm:96]: Just kill this meme pamphlet

File: code/game/machinery/vending/vendor_types/squad_prep/squad_medic.dm [hunk 2]
```diff
@@ -66,23 +66,35 @@ GLOBAL_LIST_INIT(cm_vending_gear_medic, list(
...
 		list("Whistle", 3, /obj/item/device/whistle, null, VENDOR_ITEM_REGULAR),
-		list("Welding Goggles", 3, /obj/item/clothing/glasses/welding, null, VENDOR_ITEM_REGULAR),
+
+		list("HELMET OPTICS", 0, null, null, null),
+		list("Welding Visor", 5, /obj/item/device/helmet_visor/welding_visor, null, VENDOR_ITEM_REGULAR),
+
+		list("PAMPHLETS", 0, null, null, null),
+		list("JTAC Pamphlet", 15, /obj/item/pamphlet/skill/jtac, null, VENDOR_ITEM_REGULAR),
+		list("Engineering Pamphlet", 15, /obj/item/pamphlet/skill/engineer, null, VENDOR_ITEM_REGULAR),
+		list("Powerloader Certification", 45, /obj/item/pamphlet/skill/powerloader, null, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_medic.dm:97]: Same here

File: code/game/machinery/vending/vendor_types/squad_prep/squad_rifleman.dm [hunk 3]
```diff
@@ -69,32 +69,44 @@ GLOBAL_LIST_INIT(cm_vending_clothing_marine, list(
...
-		list("Welding Goggles", 5, /obj/item/clothing/glasses/welding, null, VENDOR_ITEM_REGULAR),
+
+		list("HELMET OPTICS", 0, null, null, null),
+		list("Medical Helmet Optic", 15, /obj/item/device/helmet_visor/medical, null, VENDOR_ITEM_REGULAR),
+		list("Welding Visor", 5, /obj/item/device/helmet_visor/welding_visor, null, VENDOR_ITEM_REGULAR),
+
+		list("PAMPHLETS", 0, null, null, null),
 		list("JTAC Pamphlet", 15, /obj/item/pamphlet/skill/jtac, null, VENDOR_ITEM_REGULAR),
 		list("Engineering Pamphlet", 15, /obj/item/pamphlet/skill/engineer, null, VENDOR_ITEM_REGULAR),
 		list("Powerloader Certification", 45, /obj/item/pamphlet/skill/powerloader, null, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_rifleman.dm:109]: Same here

File: code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm [hunk 4]
```diff
@@ -23,18 +23,27 @@ GLOBAL_LIST_INIT(cm_vending_gear_smartgun, list(
...
 		list("Fulton Device Stack", 5, /obj/item/stack/fulton, null, VENDOR_ITEM_REGULAR),
+
+		list("HELMET OPTICS", 0, null, null, null),
 		list("Medical Helmet Optic", 15, /obj/item/device/helmet_visor/medical, null, VENDOR_ITEM_REGULAR),
+		list("Welding Visor", 5, /obj/item/device/helmet_visor/welding_visor, null, VENDOR_ITEM_REGULAR),
+
+		list("PAMPHLETS", 0, null, null, null),
+		list("JTAC Pamphlet", 15, /obj/item/pamphlet/skill/jtac, null, VENDOR_ITEM_REGULAR),
+		list("Engineering Pamphlet", 15, /obj/item/pamphlet/skill/engineer, null, VENDOR_ITEM_REGULAR),
+		list("Powerloader Certification", 45, /obj/item/pamphlet/skill/powerloader, null, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm:46]: Same here

File: code/game/machinery/vending/vendor_types/squad_prep/squad_specialist.dm [hunk 5]
```diff
@@ -36,17 +36,27 @@ GLOBAL_LIST_INIT(cm_vending_gear_spec, list(
...
-		list("Machete Pouch (Full)", 15, /obj/item/storage/pouch/machete/full, null, VENDOR_ITEM_REGULAR),
+
+		list("HELMET OPTICS", 0, null, null, null),
+		list("Medical Helmet Optic", 15, /obj/item/device/helmet_visor/medical, null, VENDOR_ITEM_REGULAR),
+		list("Welding Visor", 5, /obj/item/device/helmet_visor/welding_visor, null, VENDOR_ITEM_REGULAR),
+
+		list("PAMPHLETS", 0, null, null, null),
 		list("JTAC Pamphlet", 15, /obj/item/pamphlet/skill/jtac, null, VENDOR_ITEM_REGULAR),
 		list("Engineering Pamphlet", 15, /obj/item/pamphlet/skill/engineer, null, VENDOR_ITEM_REGULAR),
 		list("Powerloader Certification", 45, /obj/item/pamphlet/skill/powerloader, null, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_specialist.dm:62]: Same here

File: code/game/machinery/vending/vendor_types/squad_prep/squad_tl.dm [hunk 6]
```diff
@@ -29,25 +29,35 @@ GLOBAL_LIST_INIT(cm_vending_gear_tl, list(
...
-		list("Machete Pouch (Full)", 15, /obj/item/storage/pouch/machete/full, null, VENDOR_ITEM_REGULAR),
 		list("Fire Extinguisher (Portable)", 5, /obj/item/tool/extinguisher/mini, null, VENDOR_ITEM_REGULAR),
 		list("Whistle", 5, /obj/item/device/whistle, null, VENDOR_ITEM_REGULAR),
+
+		list("HELMET OPTICS", 0, null, null, null),
+		list("Medical Helmet Optic", 15, /obj/item/device/helmet_visor/medical, null, VENDOR_ITEM_REGULAR),
+		list("Welding Visor", 5, /obj/item/device/helmet_visor/welding_visor, null, VENDOR_ITEM_REGULAR),
+
+		list("PAMPHLET", 0, null, null, null),
 		list("Powerloader Certification", 45, /obj/item/pamphlet/skill/powerloader, null, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_tl.dm:60]: Same here

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Looks fine otherwise

File: code/game/machinery/vending/vendor_types/squad_prep/squad_leader.dm (hunk 1 above)
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_leader.dm]: removed

File: code/game/machinery/vending/vendor_types/squad_prep/squad_medic.dm (hunk 2 above)
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_medic.dm]: removed

File: code/game/machinery/vending/vendor_types/squad_prep/squad_rifleman.dm (hunk 3 above)
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_rifleman.dm]: removed

File: code/game/machinery/vending/vendor_types/squad_prep/squad_specialist.dm (hunk 5 above)
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_specialist.dm]: removed

File: code/game/machinery/vending/vendor_types/squad_prep/squad_tl.dm (hunk 6 above)
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_tl.dm]: removed

File: code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm (hunk 4 above)
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_smartgunner.dm]: removed

File: code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm [hunk 7]
```diff
@@ -46,32 +46,42 @@ GLOBAL_LIST_INIT(cm_vending_gear_engi, list(
...
 		list("Fire Extinguisher (Portable)", 3, /obj/item/tool/extinguisher/mini, null, VENDOR_ITEM_REGULAR),
 		list("Motion Detector", 8, /obj/item/device/motiondetector, null, VENDOR_ITEM_REGULAR),
 		list("Whistle", 3, /obj/item/device/whistle, null, VENDOR_ITEM_REGULAR),
+
+		list("HELMET OPTICS", 0, null, null, null),
+		list("Medical Helmet Optic", 12, /obj/item/device/helmet_visor/medical, null, VENDOR_ITEM_REGULAR),
+
+		list("PAMPHLETS", 0, null, null, null),
 		list("JTAC Pamphlet", 15, /obj/item/pamphlet/skill/jtac, null, VENDOR_ITEM_REGULAR),
 		list("Powerloader Certification", 45, /obj/item/pamphlet/skill/powerloader, null, VENDOR_ITEM_REGULAR),
```
Comment (morrowwolf) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm:82]: powerloader cert here

**Review by morrowwolf — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm (hunk 7 above)
Comment (Huffie56) [reply code/game/machinery/vending/vendor_types/squad_prep/squad_engineer.dm]: removed

**Pull request merged**